### ISSUE TITLE
[linux] Segfault Crash upon clicking "Enter As" on Hero Selection

### ISSUE THREAD

**Issue by nbaztec** [ValveSoftware/Dota-2#448]
Separated from #422 
https://github.com/ValveSoftware/Dota-2/issues/422#issuecomment-140095723

> Same happens to me, except when you click to spawn at Limited Heroes selection. Game will then crash for atleast once more while reconnecting. Can reproduce this 100% of the time.
> 
> Steps:
> Limited Heroes (on SEA server not that it matters)
> Select Hero (occasionally it won't show the model when I lock it)
> Click button to spawn/start game > Crash.

Crashlog and dumps
https://www.dropbox.com/sh/ojk7077x3yhme0t/AACiMQ1AsRwZ_UBsYmgSCEy8a?dl=0

System Information
https://gist.github.com/nbaztec/d78ebed411a60c432f15

**Comment by Tele42** [comment]
This issue should also be reported to the video driver vendor.
From the newer of the two minidumps:

```
Operating system: Linux
                  0.0.0 Linux 3.13.0-63-generic #103-Ubuntu SMP Fri Aug 14 21:42:59 UTC 2015 x86_64
CPU: amd64
     family 16 model 10 stepping 0
     6 CPUs

Crash reason:  SIGSEGV
Crash address: 0x7fbed604d000

Thread 9 (crashed)
 0  libc-2.19.so + 0x91c62
    rbx = 0x00007fbed6030000   r12 = 0x000000000000f000
    r13 = 0x00007fbf0a64a000   r14 = 0x0000000000000000
    r15 = 0x00007fbed603f000   rip = 0x00007fbf07a42c62
    rsp = 0x00007fbef872e2e8   rbp = 0x00007fbed603f000
    Found by: given as instruction pointer in context
 1  libc-2.19.so + 0x8c515
    rip = 0x00007fbf07a3d515   rsp = 0x00007fbef872e2f0
    Found by: stack scanning
 2  fglrx_dri.so + 0xe9dd81
    rip = 0x00007fbefc3bad81   rsp = 0x00007fbef872e320
    Found by: stack scanning
 3  fglrx_dri.so + 0xe9e185
    rip = 0x00007fbefc3bb185   rsp = 0x00007fbef872e3a0
    Found by: stack scanning
 4  fglrx_dri.so + 0x103df13
    rip = 0x00007fbefc55af13   rsp = 0x00007fbef872e420
    Found by: stack scanning
 5  fglrx_dri.so + 0x101107e
    rip = 0x00007fbefc52e07e   rsp = 0x00007fbef872e440
    Found by: stack scanning
 6  fglrx_dri.so + 0xec61f6
    rip = 0x00007fbefc3e31f6   rsp = 0x00007fbef872e4a0
    Found by: stack scanning
 7  fglrx_dri.so + 0x202d744
    rip = 0x00007fbefd54a744   rsp = 0x00007fbef872e508
    Found by: stack scanning
 8  fglrx_dri.so + 0x202d744
    rip = 0x00007fbefd54a744   rsp = 0x00007fbef872e568
    Found by: stack scanning
 9  fglrx_dri.so + 0x1289202
    rip = 0x00007fbefc7a6202   rsp = 0x00007fbef872e5c0
    Found by: stack scanning
10  fglrx_dri.so + 0xcbadd8
    rip = 0x00007fbefc1d7dd8   rsp = 0x00007fbef872e620
    Found by: stack scanning
11  fglrx_dri.so + 0xedb1d0
    rip = 0x00007fbefc3f81d0   rsp = 0x00007fbef872e660
    Found by: stack scanning
12  fglrx_dri.so + 0xecb999
    rip = 0x00007fbefc3e8999   rsp = 0x00007fbef872e720
    Found by: stack scanning
13  fglrx_dri.so + 0xcba341
    rip = 0x00007fbefc1d7341   rsp = 0x00007fbef872e7e0
    Found by: stack scanning
```

**Comment by nishb** [comment]
The older minidump is while clicking "Enter As", the second minidump is from the secondary crash that almost always happens when I restart+reconnect to join the game again. 
The driver is the latest one from fglrx (15.20) in trusty (14.04) repositories.

**Comment by nbaztec** [comment]
The older minidump is while clicking "Enter As", the second minidump is from the secondary crash that almost always happens when I restart+reconnect to join the game again. 
The driver is the latest one from fglrx (15.20) in trusty (14.04) repositories.

**Comment by danginsburg** [comment]
The link to your system information is not working, but from your previous post it looks like you are on AMD 7700/7800.  Can you clarify your exact GPU or repost your system info from Steam -> Help -> System Information.  I can't repro this here on Ubuntu 15.04 x64 AMD 7950 with Catalyst 15.7 (fglrx 15.20).  I tried a Limited Heroes game and it worked fine.  

Also, as @Tele42 says, the two minidumps are showing crashes inside of fglrx_dri.so.  It appears these are on threads spawned by the driver because I do not see a callstack from our application.  I would suggest filing a bug with AMD.  I would like to repro it here to see if I can narrow it down further, but so far no luck.

**Comment by danginsburg** [comment]
AMD just released Catalyst 15.9 for Linux.  Can you see if this fixes the issue for you? http://support.amd.com/en-us/download/desktop?os=Linux+x86

**Comment by nbaztec** [comment]
Fixed with the new driver  fglrx-15.201.1151 (Catalyst 15.9)
Closing.
